apps/backend/src/modules/auth/auth.controller.ts

```diff
@@ -37,6 +43,35 @@ class AuthController extends BaseController {
 		});
 	}
 
+	/**
+	 * @swagger
+	 * /auth/authenticated-user:
+	 *    get:
+	 *      description: Get authenticated user's data
+	 *      responses:
+	 *        200:
+	 *          description: Successful operation
+	 *          content:
+	 *            application/json:
+	 *              schema:
+	 *                type: object
+	 *                properties:
+	 *                  message:
+	 *                    type: object
+	 *                    $ref: "#/components/schemas/User"
+	 */
+
+	private getAuthenticatedUser(
+		options: APIHandlerOptions,
+	): Promise<APIHandlerResponse> {
+		const { user } = options;
+
+		return Promise.resolve({
+			payload: user,
+			status: HTTPCode.OK,
+		});
+	}
```

Comment:
```suggestion
		return {
			payload: user ?? null,
			status: HTTPCode.OK,
		};
	}
```

Reply:
Makes sense. But we still do not need Promise.resolve here, you can just return an object